### ISSUE TITLE
Switch to using Pipfile

### ISSUE THREAD

**Issue by acviana** [TerbiumLabs/python-matchlightsdk#38]
We might want to update this project to use a `Pipfile` to match our internal projects?

**Comment by acviana** [comment]
@mattgeiger, canI grab this issue from you? I think I have an idea on how to proceed.

**Comment by acviana** [comment]
I think there are two ways forward here; @matt and I have each taken a stab at one and we should discuss how to move forward. 

**Tox + Pipenv**
Reference: https://pipenv.readthedocs.io/en/latest/advanced/#tox-automation-project
Branch: https://github.com/TerbiumLabs/python-matchlightsdk/tree/acv-switch-to-pipenv

I'm trying to update `tox` to use `pipenv` as in the instructions above. But it's not finding the `matchlight` package which is just leaving me scratching my head. Even if we don't want to keep `tox` as our virtualization framework in the long run this would solve the core problem of moving our requirements to `pipenv` with minimal changes to the rest of the package.


**Travis CI + Pipenv**
Reference: https://pipenv.readthedocs.io/en/latest/advanced/#travis-ci
Branch: https://github.com/TerbiumLabs/python-matchlightsdk/tree/38-mdy-use-pipfile

@mattgeiger (as I understand it) is trying to drop tox completely and instead use Travis CI as our virtualization framework. As long as we are going to use Travis CI in the long run I think this is the right long-term solution but it does require rewriting not just the testing configuration but also all the build and documentation publishing directives. So this is a bigger project that would block for longer but possibly produce a better result.

**Comment by mattgeiger** [comment]
@acvienna you are right that was my approach. My stance is that we are going to move away from Tox eventually anyways. So why spend effort going for an intermediary step?

If of course we like using Tox and the rest of our current build process, that changes things.
